test/temp-sensor/Program.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace temp_sensor
+{
+    using System;
+    using System.ComponentModel.DataAnnotations;
+    using System.Threading.Tasks;
+    using common;
+    using McMaster.Extensions.CommandLineUtils;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    [Command(
+        Name = "temp-sensor",
+        Description = "tempSensor end-to-end test",
+        ExtendedHelpText = @"
+
+The following variables must be set in your environment:
+    E2E_IOT_HUB_CONNECTION_STRING
+    E2E_EVENT_HUB_ENDPOINT
+If you specify `--registry` and `--user`, the following variable must also be set:
+    E2E_CONTAINER_REGISTRY_PASSWORD
+"
+    )]
+    [HelpOption]
+    [RegistryAndUserOptionsMustBeSpecifiedTogether()]
+    class Program
+    {
+        const string DefaultAgentImage = "mcr.microsoft.com/azureiotedge-agent:1.0";
+        const string DefaultHubImage = "mcr.microsoft.com/azureiotedge-hub:1.0";
+        const string DefaultSensorImage = "mcr.microsoft.com/azureiotedge-simulated-temperature-sensor:1.0";
+
+        [Required]
```

Comment:
Base on current implementation, this code block for parameter handling will be duplicate since many of these parameters are same across tests.  We may need to further improve this.

Reply:
What I have here is appropriate for a single test. When I add more tests, I'll introduce the appropriate abstractions.